convert.go

```diff
@@ -0,0 +1,147 @@
+// Package flatfs is a Datastore implementation that stores all
+// objects in a two-level directory structure in the local file
+// system, regardless of the hierarchy of the keys.
+package flatfs
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+)
+
+func Convert(oldPath string, getOldDir ShardFunc, newPath string, getNewDir ShardFunc, out io.Writer) error {
+	old, err := New(oldPath, getOldDir, false)
+	if err != nil {
+		return err
+	}
+	new, err := New(newPath, getNewDir, false)
+	if err != nil {
+		return err
+	}
+
+	if out != nil {
+		fmt.Fprintf(out, "Getting Keys...\n")
+	}
+	res, err := old.Query(query.Query{KeysOnly: true})
+	if err != nil {
+		return err
+	}
+	entries, err := res.Rest()
```

Comment:
My math might be off, but lets say i have a 4TB repo (which, it just so happens i do). Thats 4,000,000,000,000 bytes, which if we assume the average object size is (conservatively) 8k, thats half a billion keys. That means we're going to have an array of 500,000,000 query entries. Which, if i throw some random magic math at this, comes out to something like 30GB of memory used.

So yeah, it sucks to not have an idea of how many items there are going to be, but its really not acceptable to buffer these all in memory anymore (as i know we've done in the past).

Lets change the progress meter to just show total number of things processed so far, and skip the time estimates.